### DISCUSSION TITLE
puzzles/math/2024/09/05/steve-ballmer-was-wrong

### DISCUSSION THREAD

**Discussion by giscus** [gukoff/gukoff.github.io#19] [Announcements]
# puzzles/math/2024/09/05/steve-ballmer-was-wrong

A few days ago John Graham-Cumming posted about “Steve Ballmer’s incorrect binary search interview question” which drew a lot of attention on Hacker News. The Ballmer’s favorite brain teaser goes like this:

https://gukov.dev/puzzles/math/2024/09/05/steve-ballmer-was-wrong.html

**Comment by drvinceknight** [comment]
This is a really nice write up. Thanks for sharing it :)

**Comment by possibly-wrong** [comment]
Thanks for a very interesting article! I did a similar analysis evaluating all possible search strategies up to n=13, and after reading this, ran your code to generate all of your search strategies up to n=100, evaluated them in the Mathematica code that I wrote. See the addendum and figure comparing the results here:
https://possiblywrong.wordpress.com/2024/09/04/analysis-of-adversarial-binary-search-game/
